Adjusting .sr mixins for NVDA/JAWS on Windows

clrux opened the pull request:
This work resolves issue [AC-323](https://openedx.atlassian.net/browse/AC-323) which details an issue on Windows machines using NVDA and JAWS where `.sr` text would be without spaces. For example, "Course nav menu" as the `.sr` text - using the original `.sr` class would be read as "Coursenavmenu".

After simplifying our mixins to utilize just the rules specified in the article, the text worked properly.

## Solution

There was a rogue `word-wrap: break-word` property that, when removed, allowed spaces to appear in the text. I've added `word-wrap: normal` to the `.sr` mixins which ensures that spaces within text will flow as normal without interrupting any inheritance.

As an interesting side note, removing `height` and `width` from the `.sr` mixin also seems to work. It seems to be the combination of dimension with word-wrapping. See [this JSFiddle for further examples and experimentation](https://jsfiddle.net/clrux/42atyze3/) with this.

## Platforms and browsers

This change has been tested on the following platforms and browsers:

| Platform | Version | Browser | Version | Passed |
|----------------------|---------|------------|-------|-------|
| Windows | 8 | IE | 11 | Yes |
| Windows | 8 | Firefox | 43.0 | Yes |
| Windows | 8 | Chrome | 47.0 | Yes |
| Windows | 10 | Edge | 25.1 | Yes |
| Windows | 10 | Firefox | 43.0 | Yes |
| Windows | 10 | Chrome | 47.0 | Yes |
| Mac | 10.11 | Chrome | 47.0  | Yes |
| Mac | 10.11 | Firefox | 43.0 | Yes |
| Mac | 10.11 | Safari | 9.0.1 | Yes |

## Sandbox

http://clrux-ac-323.sandbox.edx.org
http://studio-clrux-ac-323.sandbox.edx.org

## Reviewers

- [x] @cptvitamin 
- [x] @talbs or @chris-mike 

--- 

__Note: Similar work will need to be done on a different repo for the marketing site. I will follow up with someone from that team to implement the changes there and request the three of us be tagged on that PR.__

talbs commented:
@clrux, this seems reasonable and thanks for updating this utility for us. A few notes/requests:

* Can you spin up a sandbox so I can see how this change renders within our apps across browsers?
* Is the ``clip`` property supported across all of the browsers we support?
* Does this mean we can revisit the black and white contrast utilities you had to create to supplement the ``sr`` utility?
* Subsequent PRs will need to be made elsewhere (mostly in the UXPL). I think it makes sense for @andy-armstrong or @dsjen to do this as they want to be more familiar with making changes and the process of releasing/publishing. Please do not do this work yourself.

clrux commented:
@talbs Sandboxes currently reaching maximum centrifugal force.

talbs commented:
> @talbs Sandboxes currently reaching maximum centrifugal force.

You forgot to close your ``<dork>`` element above :P

clrux commented:
The W3 states that user agents should continue to support `clip` so I advise leaving it as is for the time being. I've made a note in each Sass partial that the property has been deprecated.

Initially I determined that `width` and `height` were the culprits and so I removed them from the mixins. I then discovered that the `.sr` classes were inheriting a `word-wrap: break-word` property from somewhere else. Experimentation proved that not only was this not needed on `.sr` classed elements, but that it was also causing the spaces to be removed.

I decided that, rather than removing the `word-wrap` property from higher up - as there may be other elements that rely on it - I'd overwrite it in the `.sr` mixins. I've set it to `normal`, which can be seen in the second commit. I've also replaced the previously removed properties from these mixins to "minimize the damage".

I've tested this in the browsers listed above.

clrux commented:
jenkins run lettuce

cptvitamin commented:
:+1:   I filed a bug on Mozilla and they confirmed it. https://bugzilla.mozilla.org/show_bug.cgi?id=1241631  It has to do with the FF rendering engine and how it populates the AAPI.  @clrux approach is not likely to have any impact anywhere else in the platform.

 I will write a blog post about this potential conflict in the coming days.

talbs commented:
:+1: 

FYI, @dsjen and @andy-armstrong. This change will need to be synced up in the UXPL's utilities (https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fglobal%2F_helpers.scss#L55).
